src/main/groovy/edu/oregonstate/mist/invencycle/resources/BikeResource.groovy

```diff
@@ -109,7 +144,8 @@ class BikeResource extends Resource {
         def errorMessages = ["bike_make", "bike_model", "bike_year", "bike_msrp"]
         def friendlyMessage
 
-        if (dbError.contains("cannot insert NULL into (\"MISTSTU3\".\"BIKE\".\"BIKE_BIKE_TYPE_ID\")")) {
+        if (dbError.contains("cannot insert NULL into (\"MISTSTU3\".\"BIKE\".\"BIKE_BIKE_TYPE_ID\")")
+                            || dbError.contains("cannot update (\"MISTSTU3\".\"BIKE\".\"BIKE_BIKE_TYPE_ID\") to NULL")) {
```

Comment:
This check seems fragile. It hardcodes the db name, and depends on the wording of the error message never changing. It is better to use more structured error data if possible, like SQLException.code, although it this case i'd probably just validate the bike_type property before trying to insert it.